modules/navier_stokes/src/fvkernels/INSFVTKEDSourceSink.C

```diff
@@ -112,11 +114,16 @@ INSFVTKEDSourceSink::computeQpResidual()
 
     const auto & face_info_vec = libmesh_map_find(_face_infos, _current_elem);
     const auto & distance_vec = libmesh_map_find(_dist, _current_elem);
+    mooseAssert(distance_vec.size(), "Should have found a distance vector");
+    mooseAssert(distance_vec.size() == face_info_vec.size(),
+                "Should be as many distance vectors as face info vectors");
 
     for (unsigned int i = 0; i < distance_vec.size(); i++)
     {
       const auto distance = distance_vec[i];
+      mooseAssert(distance > 0, "Should be at a non-zero distance");
```

Comment:
Is this check needed at this point? It seems to justbe adding computational burden every time that we call this kernel. Please advise

Reply:
same here. It does not hurt